pydantic_evals/pydantic_evals/datasets.py

```diff
@@ -0,0 +1,376 @@
+from __future__ import annotations as _annotations
+
+import functools
+import inspect
+import sys
+from pathlib import Path
+from typing import Any, ClassVar, Generic, Literal, NotRequired, Self, Union
+
+from pydantic._internal import _typing_extra
+
+if sys.version_info < (3, 11):
+    from exceptiongroup import ExceptionGroup
+else:
+    ExceptionGroup = ExceptionGroup
+
+import yaml
+from pydantic import BaseModel, Field, ValidationError
+from pydantic_core import to_json, to_jsonable_python
+from typing_extensions import TypedDict, TypeVar
+
+from .assessments.spec import Assessment, AssessmentFunction, AssessmentSpec, get_default_registry
+
+InputsT = TypeVar('InputsT', default=dict[str, Any])
+OutputT = TypeVar('OutputT', default=dict[str, Any])
+MetadataT = TypeVar('MetadataT', default=dict[str, Any])
```

Comment:
```suggestion
MetadataT = TypeVar('MetadataT', default=dict[str, JsonValue])
```

Reply:
🤷‍♂️ leaving as is allows you to use non-serializable stuff if you want to